### ISSUE TITLE
Enhance documentation on Azure SAS tokens

### ISSUE THREAD

**Issue by FelixBurkhard** [restic/restic#3995]
Output of `restic version`
--------------------------
restic 0.14.0 compiled with go1.19 on linux/arm

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
There should be a more detailed explanation on how to use Azure SAS tokens.
Especially the permissions restic needs to work correctly should be documented.

What are you trying to do? What problem would this solve?
---------------------------------------------------------
IMHO one of the main reasons to use a SAS token is to not give the productive system full access on the Blob Storage and to limit the time window of the access. Without the documentation the first point is hard to achieve, because everyone needs to trial & error their way through the permissions, without ever beeing sure that really everything works as intended.

I'd be happy to open a PR if someone could give me hint on where to find the required information or, if noone knows for sure, to trial & error my way through the permissions and discuss my findings here as a little sanity check.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes! I really like the easy way to setup and use restic. FOSS that just works is quite rare 😉

**Comment by rawtaz** [comment]
I just started looking into using restic with Azure and it became pretty apparent that a Secure Access Token for a Container in the blob storage, combined with a Shared Access Policy that the SAT refers to, is the way to go.

I'm not sure however that I see what you feel is missing. I mean, setting up the blob storage and getting an access token and so on, that's purely an Azure thing which presumably one has to read the Azure documentation to do. What is it you feel should be in the restic documentation?

There is an [Examples](https://restic.readthedocs.io/en/stable/080_examples.html) section in the documentation, but it's generally very risk to have elaborate tutorials in there because Azure and other providers change their GUIs, concepts and so on every now and then, so they become more or less outdated pretty quickly and need a lot of maintenance.

**Comment by FelixBurkhard** [comment]
I don't really want an detailed example on how to set up Azure. Maybe the permissions are not even something that should be documented Azure specific but more general. Obviously, restic needs permission to read and create data. But I am not sure if it needs to be able to delete or overwrite data on the repository.

**Comment by rawtaz** [comment]
> But I am not sure if it needs to be able to delete or overwrite data on the repository.

Generally speaking, restic only adds new files to the repository when you back up. But unless you use the `--no-lock` option (not recommended) it also needs to create files in the `locks/` directory, and also delete these files. Also, for operations like `forget` and `prune` restic naturally needs to be able to delete data. The same goes for `key` where you e.g. delete keys.

Regarding overwriting files this is something restic pretty much never does, since files are based on hashes of their contents new files are created instead of overwriting existing ones. One exception however is the config file.

So in summary you do need most permissions. If you are concerned about an attacker deleting files in the repository you should use features like snapshots and versioning in the blob storage, combined of course with checking your repository every now and then (before versions are removed in Azure).

**Comment by rawtaz** [comment]
@FelixBurkhard The SAS you try to use, is it an account-level SAS (Shared Access Signature, which you create in the storage account) or is it a container-level SAT (Shared Access Token, which you create in the container in the storage account)? I just discovered that container-level SAS/SAT doesn't work in #4004.

A workaround is to temporarily use an account-level SAS (or the regular account key, if one fancies that instead) for the `restic init` and after that use the container-level SAS/SAT as usual for the other restic commands (works fine since they won't try to create the container).
